### PR TITLE
[#12048] Data migration for section entities

### PR DESCRIPTION
**Pull request by mingyuanc**
Part of https://github.com/TEAMMATES/teammates/issues/12048

Outline of Solution
- Updated seedStudents to have 100 student per course and 10 section per course
- Add Migation script for section
- Add verifiction script for section

| Entity          | Test counts | Migration Time | Verification Time (Singapore) |
|-----------------|-------------|----------------|-------------------------------|
 | Section| 10000 * 10  | 1m 46s      | 10 min 10 s       |

### CODE REVIEW

**Comment by NicolasCwy** [comment]
Logic looks good but lets wait for the foreign key code to be merged

File: src/client/java/teammates/client/scripts/sql/VerifySectionAttributes.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                .distinct();
+
+    }
+
+    // Used for sql data migration
+    @Override
+    public boolean equals(teammates.storage.sqlentity.Section sqlEntity, Course datastoreEntity) {
+        return getAllSectionNames(datastoreEntity)
+                .filter(sectionName -> {
+                    try {
```
Comment (FergusMok) [inline src/client/java/teammates/client/scripts/sql/VerifySectionAttributes.java:47]: For each section in the course, we'll be loading all the course students repeatedly. It's ok for now but should be further optimized later on.

**Comment by NicolasCwy** [comment]
Could you also add the section count verification?

**Comment by NicolasCwy** [comment]
Also migration time should be non-zero, its the time taken for you to create all sections from the respective courses

File: src/client/java/teammates/client/scripts/sql/VerifySectionAttributes.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Class for verifying section attributes.
+ */
+@SuppressWarnings("PMD")
+public class VerifySectionAttributes
+        extends VerifyNonCourseEntityAttributesBaseScript<Course, teammates.storage.sqlentity.Section> {
+
+    public VerifySectionAttributes() {
+        super(Course.class,
```
Comment (FergusMok) [inline src/client/java/teammates/client/scripts/sql/VerifySectionAttributes.java:16]: Actually I realized the fix is quite fast. Could we do a quick fix and query the SQL course instead, then make sure sections are equal

**Review by FergusMok — CHANGES_REQUESTED** [review]

File: src/client/java/teammates/client/scripts/sql/VerifySectionAttributes.java (hunk 2 above)
Comment (mingyuanc) [reply src/client/java/teammates/client/scripts/sql/VerifySectionAttributes.java:20]: Done!

File: src/client/java/teammates/client/scripts/sql/VerifyCourseEntityCounts.java [hunk 3]
```diff
@@ -72,4 +79,17 @@ protected void doOperation() {
...
     }
+
+    private void verifyNewEntities() {
+        List<CourseStudent> students = ofy().load().type(CourseStudent.class).order("courseId").list();
+        verifySectionEntities(students);
+    }
+
+    private void verifySectionEntities(List<CourseStudent> students) {
+        int objectifyEntityCount = (int) students.stream().map(stu -> stu.getSectionName() + stu.getCourseId())
+                .distinct().count();
```
Comment (FergusMok) [inline src/client/java/teammates/client/scripts/sql/VerifyCourseEntityCounts.java:91]: There's 1.7m students in the database, this line may potentially cause an Out Of Memory error. But the requirements will change after today's meeting, so it'll probably work with the new design

**Review by FergusMok — APPROVED** [review]
Great work on the optimization 💯

File: src/client/java/teammates/client/scripts/sql/VerifyCourseEntityCounts.java [hunk 4]
```diff
@@ -54,6 +57,11 @@ private Long countPostgresEntities(Class<? extends teammates.storage.sqlentity.B
 
     @Override
     protected void doOperation() {
+        verifyCurrentEntities();
```
Comment (NicolasCwy) [inline src/client/java/teammates/client/scripts/sql/VerifyCourseEntityCounts.java:60]: Maybe add a javadoc comment to explain this, I can't think of a better name either

**Review by NicolasCwy — APPROVED** [review]
LGTM! One nit on naming but you can just resolve this in another PR

**Pull request merged**